central-service/controllers/room.go

```diff
@@ -0,0 +1,60 @@
+package controllers
+
+import (
+	"log"
+	"net/http"
+
+	"firebase.google.com/go/auth"
+	"github.com/acmutd/bsg/central-service/models"
+	"github.com/acmutd/bsg/central-service/services"
+	"github.com/labstack/echo/v4"
+)
+
+type RoomController struct {
+	roomService *services.RoomService
+}
+
+func InitializeRoomController(service *services.RoomService) RoomController {
+	return RoomController{service}
+}
+
+// Endpoint for creating a new room given an admin and a roomName
+func (controller *RoomController) CreateNewRoomEndpoint(c echo.Context) error {
+	var roomDTO services.RoomDTO
+	if err := c.Bind(&roomDTO); err != nil {
+		return echo.NewHTTPError(http.StatusBadRequest, "Invalid data. Please try again")
+	}
+	userAuthID := c.Get("authToken").(*auth.Token).UID
+	newRoom, err := controller.roomService.CreateRoom(userAuthID, &roomDTO)
+	if err != nil {
+		log.Printf("Failed to create room object: %v\n", err)
+		if _, ok := err.(services.RoomNameError); ok {
+			return echo.NewHTTPError(http.StatusBadRequest, "Failed to create room. " + err.Error())
+		}
+		return echo.NewHTTPError(http.StatusInternalServerError, "Failed to create room. Please try again later")
```

Comment:
Can we print out the error object before returning `http.StatusInternalServerError` so that it would be easier to debug later on?

- [ ] add error logging